### PR TITLE
[UNDERTOW-1558] security-manager and reflection permissions in Direct…

### PR DESCRIPTION
**Pull request by tmiyargi**
…ByteBufferDeallocator/undertow

Issue: https://issues.jboss.org/browse/UNDERTOW-1558

### CODE REVIEW

**Comment by undertow-pull-request** [comment]
Can one of the admins verify this patch?

**Comment by fl4via** [comment]
This is ok to test

File: core/src/main/java/io/undertow/server/DirectByteBufferDeallocator.java [hunk 1]
```diff
@@ -33,10 +33,8 @@
         Unsafe tmpUnsafe = null;
         if (version < 9) {
             try {
-                tmpCleaner = Class.forName("java.nio.DirectByteBuffer").getMethod("cleaner");
-                tmpCleaner.setAccessible(true);
-                tmpCleanerClean = Class.forName("sun.misc.Cleaner").getMethod("clean");
-                tmpCleanerClean.setAccessible(true);
+                tmpCleaner = new GetAccesibleMethod("java.nio.DirectByteBuffer", "cleaner").getMethod();
```
Comment (fl4via) [inline core/src/main/java/io/undertow/server/DirectByteBufferDeallocator.java]: For consistency purposes, can this be swapped by a "private Method getAccesibleMethod(String methodName)" that is invoked like getUnsafe0 a few lines below?

File: core/src/test/java/io/undertow/server/DirectByteBufferDeallocatorTestCase.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
```
Comment (fl4via) [inline core/src/test/java/io/undertow/server/DirectByteBufferDeallocatorTestCase.java]: Please update the license year to 2019

**Pull request merged**